### PR TITLE
Tdb/book2 img updates

### PR DESCRIPTION
**Pull request by trevordblack**
Lots of small changes in this one. Try to be thorough.

### CODE REVIEW

File: books/RayTracingTheNextWeek.html [hunk 1]
```diff
@@ -1644,14 +1653,36 @@
...
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
             double scale;
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
     };
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     [Listing [perlin-smoothed-2]: <kbd>[texture.h]</kbd> Perlin smoothed, higher frequency]
+</div>
+
+<div class='together'>
+We then add that scale to the `two_perlin_spheres()` scene description::
```
Comment (hollasch) [inline books/RayTracingTheNextWeek.html:1686]: Spurious extra colon.

File: books/RayTracingTheNextWeek.html [hunk 2]
```diff
@@ -1965,13 +2037,34 @@
...
+        // Microsoft Visual C++ Compiler
+        #pragma warning (pop)
+    #endif
+
+    #endif
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    [Listing [incl-stb-img]: Including the STB image package]
+    [Listing [rtw-stb-image]: <kbd>[rtw_stb_image.h]</kbd> Include the stb_image package]
+
+The above assumes that you have copied the `stb_image.h` into a folder called `export`. Adjust as per
```
Comment (hollasch) [inline books/RayTracingTheNextWeek.html]: “a folder called ~export~ `external`.”

Also, a suggestion: “Adjust ~as per~ according to your directory structure.”

**Comment by hollasch** [comment]
Oops, neglected the second commit...

File: books/RayTracingTheNextWeek.html [hunk 3]
```diff
@@ -2221,25 +2246,90 @@
...
     int main() {
         ...
 
-        // Render
+        point3 lookfrom;
+        point3 lookat;
+        auto vfov = 40.0;
+        auto aperture = 0.0;
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
+        color background(0,0,0);
```
Comment (hollasch) [inline books/RayTracingTheNextWeek.html:2261]: Ugh. I have no idea how things were working in my universe's timeline. Nice catch.

File: books/RayTracingTheNextWeek.html [hunk 4]
```diff
@@ -2870,7 +3045,7 @@
             constant_medium(shared_ptr<hittable> b, double d, color c)
                 : boundary(b),
                   neg_inv_density(-1/d),
-                  phase_function(make_shared<isotropic>(color(c)))
+                  phase_function(make_shared<isotropic>(make_shared<solid_color>(c)))
```
Comment (hollasch) [inline books/RayTracingTheNextWeek.html:3048]: Ooh, missed one. Like other materials, we should have an isotropic constructor that takes a color value. I'll create a separate issue for this: see #644.

**Comment by trevordblack** [comment]
I tried to also explicitly add `#include <whatever>` when clang failed to find an object type. The `includes` in the source do not exactly match the `includes` in the text. But that's not something that I want to spend time synchronizing, nor know of the ultimate value of that effort.

**Pull request merged**